### PR TITLE
Improve JDK8 corner-case test

### PR DESCRIPTION
**Pull request by hegyibalint**
I've left some unfinished work in the JDK8 test: it still relied on an external artifact, and not the locally set-up artifact.

### CODE REVIEW

File: platforms/jvm/language-java/src/integTest/groovy/org/gradle/api/tasks/compile/JavaCompileProblemsIntegrationTest.groovy [hunk 1]
```diff
@@ -304,6 +304,9 @@ class JavaCompileProblemsIntegrationTest extends AbstractIntegrationSpec impleme
     @Issue("https://github.com/gradle/gradle/pull/29141")
     @Requires(IntegTestPreconditions.Java8HomeAvailable)
     def "compiler warnings causes failure in problem mapping under JDK8"() {
+        // We are interested in how the reporting mechanism behaves when the compiler breaks, and not in the problems themselves
+        disableProblemsApiCheck()
```
[the remark this answers is not in the document]
Comment (hegyibalint) [reply platforms/jvm/language-java/src/integTest/groovy/org/gradle/api/tasks/compile/JavaCompileProblemsIntegrationTest.groovy:308]: Because the key assumption here is that we have a message arriving at the console. Do you think we - and later on everybody - to assert all problems in tests, even if the focus is not on the problems themselves?

[the remark this answers is not in the document]
Comment (hegyibalint) [reply platforms/jvm/language-java/src/integTest/groovy/org/gradle/api/tasks/compile/JavaCompileProblemsIntegrationTest.groovy:308]: Yesterday I was thinking about this, and I changed my mind: in here, we _should_ track the problems, as this is a feature which can change the layout of the problems.

In addition, realized that I've committed a mistake: I should add an unhappy path with >JDK8

[the remark this answers is not in the document]
Comment (hegyibalint) [reply platforms/jvm/language-java/src/integTest/groovy/org/gradle/api/tasks/compile/JavaCompileProblemsIntegrationTest.groovy]: Added a new test covering all other >8 JDKs

**Comment by hegyibalint** [comment]
@bot-gradle test this

**Comment by hegyibalint** [comment]
@donat I've added some extra testing, PTAL

**Review by donat — APPROVED** [review]
